src/boot/efi/pe.c

```diff
@@ -263,11 +263,19 @@ EFI_STATUS pe_memory_locate_sections(const void *base, const char * const sectio
                         sections,
                         addrs,
                         sizes,
-                        /*in_memory=*/true);
+                        in_memory);
 
         return EFI_SUCCESS;
 }
 
+EFI_STATUS pe_memory_locate_sections(const void *base, const char * const sections[], size_t *addrs, size_t *sizes) {
+        return pe_memory_locate_sections_internal(base, sections, addrs, sizes, /*in_memory=*/true);
+}
+
+EFI_STATUS pe_memory_locate_sections_offsets(const void *base, const char * const sections[], size_t *offsets, size_t *sizes) {
+        return pe_memory_locate_sections_internal(base, sections, offsets, sizes, /*in_memory=*/false);
```

Comment:
not sure how sound this is, but otherwise for a PE mule locate_sections() gives me the VMA/LMA instead of the offset, so it all goes kaboom when reading

Reply:
This is correct. `in_memory=true` is for *loaded* (for execution) images only.

In either case it's important that the image is already trusted, as we don't validate the addrs/offsets we get.